### DISCUSSION TITLE
Cannot select ai model, and API Error: Incorrect API key provided:

### DISCUSSION THREAD

**Discussion by Nova2759** [brianpetro/obsidian-smart-connections#892] [Q&A]
![Screenshot 2024-11-29 151722](https://github.com/user-attachments/assets/571f1ef6-21b0-46e4-8fe9-f81b9443f466)
![2024-11-29_15-15-44](https://github.com/user-attachments/assets/3b2f4aaa-ef6a-41c1-bb4a-2d2f73a724ad)
Not sure what is going on I have a new chat gpt API key and these errors keep showing up. Also, I for some reason cannot use the dropdown menu to select the open ai model. IT worked fine for a friend but not for me. The API key is valid too. I already restarted the app, reinstalled the plugin, and restarted my computer.

**Comment by Nova2759** [comment]
Problem got solved by generating a new key and clearing and reimporting all embedded stuff.

**Reply by brianpetro** [reply]
Good to know. Thanks for following up 🌴

**Comment by kostasvg** [comment]
Seme problem here. Generating a new key and clearing / reimporting does not solve the problem.
![Screenshot from 2024-12-05 16-58-15](https://github.com/user-attachments/assets/4853564e-97eb-40e7-a127-4740070f2052)
![Screenshot from 2024-12-05 17-02-00](https://github.com/user-attachments/assets/cc094db8-1f45-4b95-85ae-606fe258f85d)

**Reply by brianpetro** [reply]
Are there any errors in the developer console logs when you restart Obsidian? Screenshots of any errors (highlighted in red in the logs) should help uncover any issues. 

🌴

**Reply by kostasvg** [reply]
![Screenshot from 2024-12-05 21-44-18](https://github.com/user-attachments/assets/e8c88ab9-4456-4e7c-b5e9-16b67418d08d)
![Screenshot from 2024-12-05 21-46-35](https://github.com/user-attachments/assets/0e196f1a-9d8e-4e52-b574-f24cb1d797f0)
![Screenshot from 2024-12-05 21-47-10](https://github.com/user-attachments/assets/1b99f5ae-50ac-4656-a633-7845fef1470d)

**Reply by brianpetro** [reply]
Hmm. I do not see anything obvious about what may be causing the issue.

Maybe try deleting the `.smart-env` folder (in your vault folder, which requires using system file explorer to see it), then restarting Obsidian 🌴

**Reply by kostasvg** [reply]
I followed your instructions. The initial embedding process using TaylorAI/bge-micro-v2 finished okay. After that, I changed embedding model platform to OpenAI and provide an OpenAI API key. In the chat window, I also selected OpenAI and provided a different OpenAI API key. I'm getting exactly the same error messages again.

**Reply by gyeongmin-lee** [reply]
I encountered the same problem, and traced it to this piece of code.
```
/** @returns {string} OpenAI API key */
get api_key() {
    return this.settings.openai_api_key;
}
```

In order to workaround this issue, I had to add `openai_api_key` field to the `smart_env.json` file.
```
"adapter": "openai",
"api_key": "[OPENAI_API_KEY]",
"openai_api_key": "[OPENAI_API_KEY]"
```

**Reply by ravn-gt** [reply]
Hello, I am having the same issue.

**Environment:**
- Windows 11 Education build 22631
- Smart Connections Version: 2.3.40
- Obsidian Version: 1.7.7
- Using Obsidian Sync

**Problem:**
'API ket not set ...' message appears and the OpenAI embedding model does not work.

I had a similar issue previously in v2.2.76, and it was resolved by issuing a new key and applying it. However, this time the same method did not work.

**Guessing:**
![2 3 40v](https://github.com/user-attachments/assets/5fdaffa8-9ee3-4300-baee-3bb8880aecf5)
![2 3 40v-2](https://github.com/user-attachments/assets/63d8a447-6563-4764-8308-ab92fb367e81)

It seems that the cause is that Smart Connections cannot retrieve the settings.
The Smart Connections notification and settings are different if you check the image. (The Smart Connections plugin alarm is text-embedding-3-small, but the exact settings are OpenAI Text-3 Large.)

**Methods I tried:**
1. I tried adding open_api_key to smart_env.json as in the comment above(@gyeongmin-lee), but it did not work. 
2. I checked the get_api function code (line 7349-7351) in main.js of the smart embeddings plugin folder and changed open_api_key -> api_key, but it still doesn't work.

Before change)
```
7348 /** @returns {string} OpenAI API key */
7349 get api_key() {
7350     return this.settings.open_api_key;
7351 }
```

After change) The same error still occurs.
```
7348 /** @returns {string} OpenAI API key */
7349 get api_key() {
7350     return this.settings.api_key;
7351 }
```

**Reply by brianpetro** [reply]
This should be fixed in the latest version https://github.com/brianpetro/jsbrains/commit/9f7418953ff559b155feb04958c9e38b4f94bcdb

Thanks for the help debugging and bringing this to my attention 🌴

**Comment by kostasvg** [comment]
![Screenshot from 2024-12-06 14-05-58](https://github.com/user-attachments/assets/535e5d8a-ebad-4d16-9400-6288c0946269)
![Screenshot from 2024-12-06 14-06-31](https://github.com/user-attachments/assets/96a01068-293e-4997-a9fa-98396c00d729)

I tried the solution proposed by @gyeongmin-lee with no success. Using the same settings as @ravn-gt, I get a different error message.

**Reply by brianpetro** [reply]
Should be fixed in the latest version https://github.com/brianpetro/obsidian-smart-connections/discussions/892#discussioncomment-11488213 🌴

**Comment by kostasvg** [comment]
@brianpetro Thank you for the update. I've installed the update, followed by 'Clear All & Re-Import'. OpenAI is selected as Embedding Model Platform. However, the notification says 'TaylorAI/bge-micro-v2' is used for the embedding:

![Screenshot from 2024-12-06 21-57-06](https://github.com/user-attachments/assets/33b640d7-ce8a-45aa-a94d-f9dc516c4ccd)

**Reply by brianpetro** [reply]
@kostasvg I just figured this out. Fortunately, it should use the correct model (OpenAI text large in this case).

The issue is in the variable being used in the notification. The fix will be included in the next update.

Thanks for bringing this to my attention and sorry for the confusion
🌴
